### PR TITLE
Update rhods rules to track kfnbc on openshift monitoring

### PR DESCRIPTION
**Pull request by atheo89**
This PR solves the issue regarding that the Openshift Monitoring didn't display usage data for Jupyter notebooks.
More details here: https://github.com/opendatahub-io/odh-manifests/issues/662

### Changes
- Changed the pod name of the filtering from `pod=~"jupyterhub-nb.*"` to `pod=~"jupyter-nb.*"`
- In some expressions added an extra filter to distinguish the notebook from the `oauth-proxy` container with the following filter: `container=~"jupyter-nb-.*"`

### Evaluation

1. For example, to review cpu consumption on openshift metric dashboard you should run: `cluster:usage:consumption:rhods:cpu:seconds:rate1h`
2. Or, on the Prometheus dashboard under `openshift-monitoring` run a rule expression for example the following: `sum(rate(container_cpu_usage_seconds_total{container="",pod=~"jupyter-nb.*",namespace="rhods-notebooks"}[1h]))`

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] For commits that came from upstream, `[UPSTREAM]` has been prepended to the commit message
- [x] JIRA link(s): https://issues.redhat.com/browse/RHODS-4228
- [x] Live Build:  quay.io/accorvin/rhods-operator-live-catalog:1.0.0-rhods-4765
- [ ] The Jira story is acked
- [ ] An entry has been added to the latest build document in [Build Announcements Folder](https://drive.google.com/drive/folders/1sgkK1WZgGo9CXsLizNe0GbAzVKuSKrZL).
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious)

### CODE REVIEW

**Review by samuelvl — APPROVED** [review]
/lgtm

**Comment by accorvin** [comment]
I don't think we should make this change.

We are limited in the number of these rules that we create, so we need to be deliberate in when we add them. The existing rules (a little higher in the file) were deliberately picked and are used by other systems for measuring consumption for billing purposes, so we should stick with using those metric names.

I think the best thing to do is to update the existing rules to track based on kfnbc usage when we make the switch from jupyterhub to kfnbc, or possibly change the rules to try to track both (but that will probably be a fairly complex query, so I'd vote against this option).

**Comment by atheo89** [comment]
Hi @accorvin these rules are only for 1.15 testing purpose and they will be removed in 1.16 with the new ones

**Comment by atheo89** [comment]
The old rhods rules are converted to be compatible with 1.16

File: monitoring/base/rhods-rules.yaml [hunk 1]
```diff
@@ -11,14 +11,14 @@ spec:
...
     rules:
     - record: cluster:usage:consumption:rhods:cpu:seconds:rate1h
-      expr: sum(rate(container_cpu_usage_seconds_total{container="",pod=~"jupyterhub-nb.*",namespace="rhods-notebooks"}[1h]))
+      expr: sum(rate(container_cpu_usage_seconds_total{container="",pod=~"jupyter-nb.*",namespace="rhods-notebooks"}[1h]))
     - record: cluster:usage:consumption:rhods:pod:up
-      expr: count(kube_pod_container_status_ready{namespace="rhods-notebooks",pod=~"jupyterhub-nb.*"}==1)
+      expr: count(kube_pod_container_status_ready{namespace="rhods-notebooks", pod=~"jupyter-nb.*",container=~"jupyter-nb-.*[^kube+-3aadmin]"}==1)
     - record: cluster:usage:consumption:rhods:active_users
-      expr: count(count(count_over_time(kube_pod_container_status_ready{namespace="rhods-notebooks",pod=~"jupyterhub-nb.*"}[1h])) without (uid))
+      expr: count(count(count_over_time(kube_pod_container_status_ready{namespace="rhods-notebooks",pod=~"jupyter-nb.*", container=~"jupyter-nb-.*[^kube+-3aadmin]"}[1h])) without (uid))
```
Comment (accorvin) [inline monitoring/base/rhods-rules.yaml:18]: I think we can replace this ugly expression with the runningNotebooks metric that KFNBC provides.

**Comment by accorvin** [comment]
A new live build with these changes is available at quay.io/accorvin/rhods-operator-live-catalog:1.0.0-rhods-4765

**Comment by LaVLaS** [comment]
@atheo89 Is this supposed to be merged into `master` branch?

**Comment by LaVLaS** [comment]
Rhods livebuild for these changes with KFNBC and migration script: [quay.io/llasmith/rhods-operator-live-catalog:1.16.0-dashboard-kfnbc](https://quay.io/llasmith/rhods-operator-live-catalog:1.16.0-dashboard-kfnbc)

**Comment by atheo89** [comment]
> @atheo89 Is this supposed to be merged into `master` branch? I think yes because updates an old functionality.

File: monitoring/base/rhods-rules.yaml (hunk 1 above)
Comment (atheo89) [reply monitoring/base/rhods-rules.yaml]: Seems that the rule record cluster:usage:consumption:rhods:active_users doesn't work properly, with the count(notebook_running) expression. 

On the other hand the old expression is working fine: count(count(count_over_time(kube_pod_container_status_ready{namespace="rhods-notebooks",pod=~"jupyter-nb.*", container=~"jupyter-nb-.*"}[1h])) without (uid))

I will investigate it further

Comment (atheo89) [reply monitoring/base/rhods-rules.yaml]: This metric `count(kube_statefulset_replicas{namespace=~"rhods-notebooks", statefulset=~"jupyter-nb-.*"} ==1)` works better for the `cluster:usage:consumption:rhods:active_users`

**Comment by LaVLaS** [comment]
@atheo89 Yes this should be merged into the `master` branch

**Comment by LaVLaS** [comment]
I have pulled the commits from this PR into #238 so that the changes are against the `master` branch

**Comment by accorvin** [comment]
Closing this since the changes have been merged to #238

**Pull request closed without merging**